File: common-api/app/schemas/facebook_post.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.schemas.facebook_profile import FacebookProfile, FacebookProfileResponse


class FacebookPostBase(BaseModel):
    profile_id: UUID
    post_id: str
    message: str | None = None
    type: str | None = None
    link: str | None = None
    media_url: str | None = None
    media_type: Literal["image", "video"] | None = None
    status: str | None = None
    published_at: datetime | None = None


class FacebookPostCreate(FacebookPostBase):
    pass


class FacebookPostUpdate(BaseModel):
    profile_id: UUID | None = None
    message: str | None = None
    type: str | None = None
    link: str | None = None
    media_url: str | None = None
    media_type: Literal["image", "video"] | None = None
    status: str | None = None
    published_at: datetime | None = None
    deleted_at: datetime | None = None


class FacebookPostResponse(FacebookPostBase):
    id: UUID
    created_at: datetime
    updated_at: datetime | None = None
    deleted_at: datetime | None = None
    profile: FacebookProfileResponse | None = None
    model_config = ConfigDict(from_attributes=True)


class FacebookPost(FacebookPostResponse):
    profile: FacebookProfile | None = None

File: web/src/types/api/facebook-post.ts
export type FacebookPost = {
  profile_id: string;
  post_id: string;
  message?: string;
  link?: string;
  media_url?: string;
  media_type?: 'image' | 'video';
  status: 'active' | 'inactive';
  published_at: string;
};

export type FacebookPostUpdate = Partial<
  Omit<FacebookPost, 'post_id' | 'profile_id' | 'status' | 'published_at'>
> & {
  profile_id?: string;
  message?: string;
  link?: string;
  media_url?: string;
  media_type?: 'image' | 'video';
  status?: 'active' | 'inactive';
  published_at?: string;
  deleted_at?: string;
};

export type FacebookPostResponse = FacebookPost & {
  id: string;
  created_at: string;
  updated_at?: string;
  deleted_at?: string;
};
